src/main/java/com/example/bookapp/controller/BookController.java

```diff
@@ -0,0 +1,51 @@
+package com.example.bookapp.controller;
+
+import com.example.bookapp.dto.request.CreateBookRequestDto;
+import com.example.bookapp.dto.response.BookResponseDto;
+import com.example.bookapp.service.BookService;
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/books")
+@RequiredArgsConstructor
+public class BookController {
+    private final BookService bookService;
+
+    @Operation(
+            summary = "Create new book",
+            description = "Create new book from given JSON"
+    )
+    @PostMapping
+    public ResponseEntity<BookResponseDto> save(
+            @RequestBody
+            @Valid
+            CreateBookRequestDto createBookRequestDto
+    ) {
+        return ResponseEntity.ok(bookService.save(createBookRequestDto));
+    }
```

Comment:
redundant use of response entity. fix rest methods the same way
```suggestion
    public BookResponseDto save(
            @RequestBody
            @Valid
            CreateBookRequestDto createBookRequestDto
    ) {
        return bookService.save(createBookRequestDto);
    }
```